src/parser/jobs/mnk/modules/evaluators/BlitzEvaluator.tsx

```diff
@@ -100,17 +107,8 @@ export class BlitzEvaluator implements WindowEvaluator {
 	}
 
 	private calculateRechargeTimes(): void {
-		let isOnCooldown = false
-		for (let i = 0; i < this.pbCasts.length; i++) {
-			if (i !== 0) {
-				isOnCooldown = this.pbCasts[i] < this.rechargeTimes[i-1]
-			}
-
-			if (!isOnCooldown) {
-				this.rechargeTimes[i] = this.pbCasts[i] + this.pbCooldown
-			} else {
-				this.rechargeTimes[i] = this.rechargeTimes[i-1] + this.pbCooldown
-			}
+		if (this.pbChargeHistory.length === 0) {
+			this.pbChargeHistory = this.cooldowns.chargeHistory('PERFECT_BALANCE')
```

Comment:
I *think* this should work with `this.cooldowns.chargeHistory(this.data.actions.PERFECT_BALANCE)` if I'm reading the TS correctly in the Cooldowns module, which (at least in my book) would be cleaner and safer than using the string key. That said, I do see some precedent elsewhere in these files for using a string instead of the action, so you're welcome to take this with a grain of salt if it's a standard practice.

Reply:
Yeah, I like that better. Have to pass it through from the RoF analyser since evaluators don't have direct access to `data` but still.